src/resource.js

```diff
@@ -0,0 +1,213 @@
+import EventEmitter from 'events'
+import url from 'url'
+import fs from 'fs'
+import http from 'http'
+import Stream from 'stream'
+import _ from 'lodash'
+import parse from 'csv-parse'
+import transform from 'stream-transform'
+import Schema from './schema'
+import constraints from './constraints'
+import utilities from './utilities'
+
+/**
+ * @returns Promise
+ */
+export default class Resource {
+  constructor(schema, source) {
```

Comment:
In py libs we use `data` instead of `source`. May be `source` works better for urls etc but overall I suppose `schema-data` pair works better because of usage consistency in different places (when source could be not fully obvious).